packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -6,96 +6,104 @@ import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+    /**
+     * @notice The number of the most recent L2 block recorded in this contract.
+     */
+    uint256 public latestBlockNumber;
 
-    /// @notice A mapping from L2 timestamps to the output root for the block with that timestamp.
+    /**
+     * @notice A mapping from L2 block numbers to the respective output root.
+     */
     mapping(uint256 => OutputProposal) internal l2Outputs;
 
-    /// @notice OutputProposal represents a commitment to the L2 state.
-    /// The timestamp is the L1 timestamp that the output root is posted.
-    /// This timestamp is used to verify that the finalization period
-    /// has passed since the output root was submitted.
-    struct OutputProposal {
-        bytes32 outputRoot;
-        uint256 timestamp;
-    }
-
     /***************
      * Constructor *
      ***************/
 
     /**
      * @notice Initialize the L2OutputOracle contract.
-     * @param _submissionInterval The desired interval in seconds at which
-     *        checkpoints must be submitted.
-     * @param _l2BlockTime The desired L2 inter-block time in seconds.
-     * @param _genesisL2Output The initial L2 output of the L2 chain.
+     *
+     * @param _submissionInterval    The desired interval in seconds at which
+     *                               checkpoints must be submitted.
+     * @param _genesisL2Output       The initial L2 output of the L2 chain.
      * @param _historicalTotalBlocks The number of blocks that preceding the
-     *        initialization of the L2 chain.
-     * @param _startingBlockTimestamp The timestamp to start L2 block at.
+     *                               initialization of the L2 chain.
+     * @param _startingBlockNumber   The number to start L2 block at.
      */
     constructor(
         uint256 _submissionInterval,
-        uint256 _l2BlockTime,
         bytes32 _genesisL2Output,
         uint256 _historicalTotalBlocks,
-        uint256 _startingBlockTimestamp,
+        uint256 _startingBlockNumber,
         address sequencer
     ) {
-        require(
-            _submissionInterval % _l2BlockTime == 0,
-            "Submission Interval must be a multiple of L2 Block Time"
-        );
-
         SUBMISSION_INTERVAL = _submissionInterval;
-        L2_BLOCK_TIME = _l2BlockTime;
-        // solhint-disable-next-line not-rely-on-time
-        l2Outputs[_startingBlockTimestamp] = OutputProposal(_genesisL2Output, block.timestamp);
+        l2Outputs[_startingBlockNumber] = OutputProposal(_genesisL2Output, block.timestamp);
         HISTORICAL_TOTAL_BLOCKS = _historicalTotalBlocks;
-        // solhint-disable-next-line not-rely-on-time
-        latestBlockTimestamp = _startingBlockTimestamp;
-        // solhint-disable-next-line not-rely-on-time
-        STARTING_BLOCK_TIMESTAMP = _startingBlockTimestamp;
+        latestBlockNumber = _startingBlockNumber;
+        // TODO: I think we can just remove this now. It's no longer needed in computeL2BlockNumber.
```

Comment:
I'm working on this in a new PR: https://github.com/ethereum-optimism/optimism/pull/2735